06_fixation_pupil_diameter.ipynb

```diff
@@ -0,0 +1,264 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Tutorial 06 - Pupil Diameter By Fixation On Surface\n",
+    "In this tutorial we will look at how to calculate the mean pupil diameter for each fixation on a given surface.\n",
+    "We will follow these steps:\n",
+    "1. Define the necessary file paths\n",
+    "1. Load the data using `Pandas`.\n",
+    "1. Select required columns\n",
+    "1. Group fixations by id\n",
+    "1. Extract pupil data for each fixation\n",
+    "1. Calculate mean pupil diameter for based on the extracted pupil data\n",
+    "\n",
+    "Please check the documentation of [Pupil Player](https://docs.pupil-labs.com/#pupil-player) to see how to export data. \n",
+    "This Notebook is accompanied by an example [recording](https://drive.google.com/open?id=119Xxe1eYEgmXQTUK1RyytrBKly_vU6dQ) which was already exported. To re-execute the cells below, please first download the recording and unpack it inside the `recordings` directory next to the notebook.\n",
+    "\n",
+    "This tutorial expects the recording to be exported with [Pupil Player `v1.15`](https://github.com/pupil-labs/pupil/releases/tag/v1.15).\n",
+    "\n",
+    "## 1 - Define the necessary file paths\n",
+    "Let's start by importing the necessary Python modules and defining the necessary file paths."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pathlib\n",
+    "\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import seaborn as sns\n",
+    "sns.set(context=\"notebook\", style=\"whitegrid\", font_scale=1.2)\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.colors as colors\n",
+    "%matplotlib inline"
```

Comment:
Question: what is `%matplotlib inline` - what does this do?

Reply:
Short answer: It causes any matplotlib plots to be rendered within the notebook cells instead of an external window.
Long answer: https://stackoverflow.com/a/43028034/5859392

Reply:
thanks for the context.